scripts/abi_check.py

```diff
@@ -0,0 +1,240 @@
+#!/usr/bin/env python3
+#
+# This file is part of Mbed TLS (https://tls.mbed.org)
+#
+# Copyright (c) 2018, Arm Limited, All Rights Reserved
+#
+# Purpose
+#
+# This script is a small wrapper around the abi-compliance-checker and
+# abi-dumper tools, applying them to compare the ABI and API of the library
+# files from two different Git revisions within an Mbed TLS repository.
+# The results of the comparison are formatted as HTML and stored at
+# a configurable location. Returns 0 on success, 1 on ABI/API non-compliance,
+# and 2 if there is an error while running the script.
+# Note: must be run from Mbed TLS root.
+
+import os
+import sys
+import traceback
+import shutil
+import subprocess
+import argparse
+import logging
+import tempfile
+
+
+class AbiChecker(object):
+
+    def __init__(self, report_dir, old_rev, new_rev, keep_all_reports):
+        self.repo_path = "."
+        self.log = None
+        self.setup_logger()
+        self.report_dir = os.path.abspath(report_dir)
+        self.keep_all_reports = keep_all_reports
+        self.should_keep_report_dir = os.path.isdir(self.report_dir)
+        self.old_rev = old_rev
+        self.new_rev = new_rev
+        self.mbedtls_modules = ["libmbedcrypto", "libmbedtls", "libmbedx509"]
+        self.old_dumps = {}
+        self.new_dumps = {}
+        self.git_command = "git"
+        self.make_command = "make"
+
+    def check_repo_path(self):
+        if __file__ not in [os.path.join(".", "scripts", "abi_check.py"),
```

Comment:
It is not OS independent way. ```__file__``` is in OS specific form i.e. with back slashes on Windows. ```os.path.join``` by default generates POSIX style path i.e. with forward slashes. Moreover, there are different ways the script can be run from mbedtls root but not match checked paths.
One OS independent way is to find current path and script path. Adjust script path to be mbedtls root and then compare. Example:
```
cur_dir = os.path.realpath('.')
root_dir = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))

if cur_dir != root_dir:
    raise Exception("Must be run from Mbed TLS root")
```

Reply:
This is a better method, I've made the change.